psi4/src/psi4/dlpno/mp2.h

```diff
@@ -0,0 +1,180 @@
+    std::vector<std::vector<int>> i_j_to_ij; ///< LMO indices (i, j) to LMO pair index (ij)
+    std::vector<std::pair<int,int>> ij_to_i_j; ///< LMO pair index (ij) to both LMO indices (i, j)
+    std::vector<int> ij_to_ji; ///< LMO pair index (ij) to LMO pair index (ji)
+
+    // LMO Pair Domains
+    SparseMap lmopair_to_ribfs; ///< which aux BFs are needed for density-fitting a pair of LMOs?
+    SparseMap lmopair_to_riatoms; ///< aux BFs on which atoms are needed for density-fitting a pair of LMOs?
+    SparseMap lmopair_to_paos; ///< which PAOs span the virtual space of a pair of LMOs?
+    SparseMap lmopair_to_paoatoms; ///< PAOs on which atoms span the virtual space of a pair of LMOs?
+
+    // Extended LMO Domains 
+    SparseMap lmo_to_riatoms_ext; ///< aux BFs on which atoms are needed for density-fitting a LMO and all connected LMOs
+    SparseMap riatom_to_lmos_ext; ///< the extended DF domains of which LMOs include aux BFs on an atom
+    SparseMap riatom_to_paos_ext; ///< the extended DF domains of which PAOs include aux BFs on an atom
+    SparseMap riatom_to_atoms1; ///< orbital BFs on which atoms are needed for DF int transform (first index)
+    SparseMap riatom_to_shells1; ///< which shells of orbital BFs are needed for DF int transform (first index)
+    SparseMap riatom_to_bfs1; ///< which orbital BFs are needed for DF int transform (first index)
+    SparseMap riatom_to_atoms2; ///< orbital BFs on which atoms are needed for DF int transform (second index)
+    SparseMap riatom_to_shells2; ///< which shells of orbital BFs are needed for DF int transform (second index)
+    SparseMap riatom_to_bfs2; ///< which orbital BFs are needed for DF int transform (second index)
+
+    // Dense analogues of some sparse maps for quick lookup
+    std::vector<std::vector<int>> riatom_to_lmos_ext_dense;
+    std::vector<std::vector<bool>> riatom_to_atoms1_dense;
+    std::vector<std::vector<bool>> riatom_to_atoms2_dense;
+
+    void common_init();
+
+    std::pair<SharedMatrix, SharedVector> orthocanonicalizer(SharedMatrix S, SharedMatrix F);
+
+    /// form LMOs, PAOs, etc.
+    void setup();
+    
+    /// compute differential overlap integrals between LMO/LMO and LMO/PAO pairs (EQ 4), DOI_ij and DOI_iu
+    void overlap_ints();
+
+    /// compute approximate MP2 pair energies for distant LMOs using dipole integrals (EQ 17)
+    void dipole_ints();
+
+    /// assess sparsity relationships between LMOs, PAOs, RI basis, and orbital basis
+    void sparsity_prep();
+
+    /// compute three-index integrals in LMO/PAO basis with linear scaling (EQ 11)
+    void df_ints();
+
+    /// form pair exch operators (EQ 15) and SC amplitudes (EQ 18); transform to PNO basis
+    void pno_transform();
+
+    /// compute PNO/PNO overlap matrices
+    void pno_overlaps();
+
+    /// compute MP2 correlation energy w/ current amplitudes (EQ 14)
+    double eval_amplitudes();
```

Comment:
Rename this function. When I read this, I thought the amplitudes are the _output_ of your evaluation, not the _input_. Might I suggest `compute_energy_from_amplitudes` or `compute_iteration_energy`?